meshtastic/interfaces/ble/lifecycle_receive_runtime.py

```
@@ -0,0 +1,139 @@
+
+    def set_receive_wanted(self, *, want_receive: bool) -> None:
+        """Request or clear receive-loop intent."""
+        with self._iface._state_lock:
+            self._iface._want_receive = want_receive
+
+    def should_run_receive_loop(self) -> bool:
+        """Return whether receive loop should continue running."""
+        with self._iface._state_lock:
+            return self._iface._want_receive and not self._iface._closed
+
+    def start_receive_thread(
```

Ruff (0.15.6)

[warning] Too many branches (16 > 12)

(PLR0912)